util/settings/resources_filter.go

```diff
@@ -64,13 +64,23 @@ func (rf *ResourcesFilter) isExcludedResource(apiGroup, kind, cluster string) bo
 // +-------------+-------------+-------------+
 //
 func (rf *ResourcesFilter) IsExcludedResource(apiGroup, kind, cluster string) bool {
-	if len(rf.ResourceInclusions) > 0 {
-		if rf.isIncludedResource(apiGroup, kind, cluster) {
-			return rf.isExcludedResource(apiGroup, kind, cluster)
-		} else {
+	// if excluded, do not allow
+	if rf.isExcludedResource(apiGroup, kind, cluster) {
+		return true
+	}
+
+	// if included, do allow
+	if rf.isIncludedResource(apiGroup, kind, cluster) {
+		return false
+	}
+
+	// if inclusion rules defined for cluster, default is not allow
+	for _, includedResource := range rf.ResourceInclusions {
+		if includedResource.MatchCluster(cluster) {
```

Comment:
Hi @darshanime, why can't we retain `includesResource.Match()` here since it uses `matchCluster()` internally?

Reply:
@chetan-rns `includeResource.Match()` [also matches](https://github.com/argoproj/argo-cd/pull/4027/files#diff-5c2e32297eb471444576d268b3992d53R39) `apiGroup` and `kind`, which we don't want here.   
We are just trying to decide if there are existing whitelisting rules that apply to this cluster, so we can decide the default policy.